packages/relayer/processor/process_message.go

```diff
@@ -461,3 +451,48 @@ func (p *Processor) saveMessageStatusChangedEvent(
 
 	return nil
 }
+
+func (p *Processor) setGasTipOrPrice(ctx context.Context, auth *bind.TransactOpts) error {
+	gasTipCap, err := p.destEthClient.SuggestGasTipCap(ctx)
+	if err != nil {
+		if IsMaxPriorityFeePerGasNotFoundError(err) {
+			auth.GasTipCap = FallbackGasTipCap
+		} else {
+			gasPrice, err := p.destEthClient.SuggestGasPrice(context.Background())
+			if err != nil {
+				return errors.Wrap(err, "p.destBridge.SuggestGasPrice")
+			}
+			auth.GasPrice = gasPrice
+		}
+	} else {
+		auth.GasTipCap = gasTipCap
+
+		return nil
+	}
+
+	return nil
+}
+
+func (p *Processor) getCost(ctx context.Context, auth *bind.TransactOpts) (*big.Int, error) {
+	if auth.GasTipCap != nil {
+		bn, err := p.destEthClient.BlockNumber(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		blk, err := p.destEthClient.BlockByNumber(context.Background(), new(big.Int).SetUint64(bn))
+		if err != nil {
+			return nil, err
+		}
+
+		cfg := params.NetworkIDToChainConfigOrDefault(p.destChainId)
+
+		baseFee := eip1559.CalcBaseFee(cfg, blk.Header())
```

Comment:
whats the difference between this and `blk.BaseFee()`?

Reply:
eip1559.CalcBaseFee calculates basefee based on gasUsed of the previous block

Reply:
emmm for L2 => L1 relayer i think this will work, but for L1 => L2 relayer, u should try `TaikoL2.getBasefee`?